zbd.c

```diff
@@ -1283,8 +1295,24 @@ static void zbd_put_io(const struct io_u *io_u)
 	       "%s: terminate I/O (%lld, %llu) for zone %u\n",
 	       f->file_name, io_u->offset, io_u->buflen, zone_idx);
 
-	ret = pthread_mutex_unlock(&z->mutex);
-	assert(ret == 0);
+	if (td->o.zone_append && z->pending_ios > 0) {
+		pthread_mutex_lock(&z->mutex);
+		z->pending_ios--;
+		/*
+		 * Other threads may be waiting for pending I/O's to
+		 * complete for this zone. Notify them.
+		 */
+		if (!z->pending_ios)
+			pthread_cond_broadcast(&z->reset_cond);
+		ret = pthread_mutex_unlock(&z->mutex);
+		assert(ret == 0);
+	}
+
+	if (!td->o.zone_append) {
+		ret = pthread_mutex_unlock(&z->mutex);
+		assert(ret == 0);
+	}
```

Comment:
The previous hunk could be moved here as "} else if (z->pending_ios > 0) {". That will simplify reading and reduce condition tests.

Reply:
Sure, this code will be refactored in the v2 patches.